spec/models/satisfaction_survey_feature_metrics_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+RSpec.describe SatisfactionSurveyFeatureMetrics, with_audited: true do
+  subject(:feature_metrics) { described_class.new }
+
+  def create_application
+    application_form = create(:completed_application_form, submitted_at: Time.zone.now)
+    application_form
+  end
+
+  def create_application_with_feeback
+    application_form2 = create(:completed_application_form, :with_feedback_completed, submitted_at: Time.zone.now)
+    application_form2
+  end
+
+  describe '#formatted_response_rate' do
+    context 'without any data' do
+      it 'returns n/a' do
+        expect(feature_metrics.formatted_response_rate(1.month.ago)).to eq('n/a')
+      end
+    end
+
+    context 'with applications with feedback' do
+      it 'returns 0 when there are no applications with feedback' do
+        create_application
+        expect(feature_metrics.formatted_response_rate(1.month.ago)).to eq('0%')
+      end
+
+      it 'returns the right percentages when applications with feedback exist' do
+        create_application
+        2.times { create_application_with_feeback }
+        expect(feature_metrics.formatted_response_rate(1.month.ago)).to eq('66.7%')
+      end
+
+      it 'returns the right percentages over a range of dates' do
```

Comment:
This test block is inspired by some adjacent objects used by the dashboard. The aim is to cover some of the functionality missing from the integration test. It would be more expensive to replicate this in that spec. 

However, I may have missed the point here so open to any thoughts.

Reply:
This looks good to me.

I think the intention here was just to cover some of the query logic for time intervals. 

Using Timecop is a bit clumsy. I couldn't think of a better solution for some of the other metrics mainly because there were conditions on things in the audit trail that the conditions depended on and faking those out would have been worse. In this case the query condition only depends on `ApplicationForm#submitted_at` so you could explicitly set values for each application form that you create. That might be tidier but it's consistent the way it is so don't mind either way.